### PR TITLE
docs: add NetworkPolicy examples to security best practices

### PR DESCRIPTION
**Pull request by swiatekm-sumo**
##### Description

Add some example Network Policies illustrating how to limit collection components to only necessary networking capabilities.

### CODE REVIEW

File: deploy/docs/Security_Best_Practices.md [hunk 1]
```diff
@@ -39,6 +39,52 @@ requests over the network.
...
+kind: NetworkPolicy
+metadata:
+  name: fluent-bit
+spec:
+  podSelector:
+    matchLabels:
+      app.kubernetes.io/name: fluent-bit
+  policyTypes:
+  - Ingress
+  - Egress
```
Comment (pmalek-sumo) [inline deploy/docs/Security_Best_Practices.md:55]:
```suggestion
  policyTypes:
  - Egress
```

File: deploy/docs/Security_Best_Practices.md [hunk 2]
```diff
@@ -55,6 +157,47 @@ be able to reach the collector over the network.
...
+    - protocol: TCP
+      port: 14268
+    - protocol: TCP
+      port: 55678
+    - protocol: TCP
+      port: 55680
+    - protocol: TCP
+      port: 55681
+  egress: 
+  - {}
```
Comment (pmalek-sumo) [inline deploy/docs/Security_Best_Practices.md:195]: What's this for?

File: deploy/docs/Security_Best_Practices.md (hunk 1 above)
Comment (swiatekm-sumo) [reply deploy/docs/Security_Best_Practices.md:55]: This is intentional, adding `Ingress` here and then not providing a configuration for it means "reject all inbound connections.

File: deploy/docs/Security_Best_Practices.md (hunk 2 above)
Comment (swiatekm-sumo) [reply deploy/docs/Security_Best_Practices.md:195]: It means "allow all outbound connections". It's basically here for completeness' sake.

File: deploy/docs/Security_Best_Practices.md (hunk 1 above)
Comment (swiatekm-sumo) [reply deploy/docs/Security_Best_Practices.md:55]: I feel like if someone is actually following this guide, they should already know how Network Policies work.

Comment (pmalek-sumo) [reply deploy/docs/Security_Best_Practices.md:55]: People might also learn stuff from our docs :)

Comment (swiatekm-sumo) [reply deploy/docs/Security_Best_Practices.md:55]: I realized we should actually allow trafic on Fluent-Bit's metrics port, so I added it here along with a comment.

**Comment by swiatekm-sumo** [comment]
I decided to add comments to all the rules, hopefully that'll make it clearer.

**Review by pmalek-sumo — APPROVED** [review]

**Pull request merged**